### PR TITLE
sys_memery fix

### PR DESCRIPTION
**Pull request by isJuhn**
Add null alloc_addr checks to sys_memory_allocate and sys_memory_allocate_from_container.
[testcase](https://github.com/isJuhn/ps3tests/blob/master/sys_memory_allocate/expected.txt)
Some game was doing this according to ani.

### CODE REVIEW

**Comment by AniLeo** [comment]
Fixes the following issue with Ridge Racer 7 [NPUB30457] (and other games that try to call those functions with nullptr as alloc_addr)

```
·W 0:07:55.795840 {PPU[0x100004d] Thread (bnusCoreDecoderAT3PThread) [0x00399d00]} sys_memory: sys_memory_allocate(size=0xe0000, flags=0x200, alloc_addr=*0x0)
·F 0:07:55.796045 {PPU[0x100004d] Thread (bnusCoreDecoderAT3PThread) [0x00399d00]} MEM: Access violation writing location 0x0
```

File: rpcs3/Emu/Cell/lv2/sys_memory.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#include "stdafx.h"
+﻿#include "stdafx.h"
```
Comment (janisozaur) [inline rpcs3/Emu/Cell/lv2/sys_memory.cpp]: Why is BOM necessary here?

Comment (MSuih) [reply rpcs3/Emu/Cell/lv2/sys_memory.cpp]: This project uses BOM for some reason
https://github.com/RPCS3/rpcs3/blob/0e0a82e5363cabdfd417d907847232ececb4fdd5/.editorconfig#L4
It's stupid, but that's what the project has decided to do.

Comment (ruipin) [reply rpcs3/Emu/Cell/lv2/sys_memory.cpp]: This was decided after multiple contributors had issues where an editor or the compiler didn't auto-detect UTF-8 properly and broke stuff such as log messages.

File: rpcs3/Emu/Cell/lv2/sys_memory.cpp [hunk 2]
```diff
@@ -43,6 +45,12 @@ error_code sys_memory_allocate(u32 size, u64 flags, vm::ptr<u32> alloc_addr)
 		return CELL_ENOMEM;
 	}
 
+	if (!alloc_addr)
+	{
+		dct->used -= size;
+		return {CELL_EFAULT, alloc_addr};
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/lv2/sys_memory.cpp]: For the sake of nitpicking, it's better to not use second argument because it's clearly zero.

**Pull request merged**